---
title: Introduction to _de novo_ Genome Assembly
author: Alexander HÃ¼bner
---

```{r echo = F, message = F}
library(tidyverse)
library(pander)
```


::: {.callout-tip}
For this chapter's exercises, if not already performed, you will need to create the [conda environment](before-you-start.qmd#creating-a-conda-environment) from the `yml` file in the following [link](https://github.com/SPAAM-community/intro-to-ancient-metagenomics-book/raw/main/assets/envs/denovo-assembly.yml) (right click and save as to download), and once created, activate the environment with:

```bash
conda activate denovo-assembly
```
:::

### Lecture

Lecture slides and video from the [2022 edition of the summer school](https://www.spaam-community.org/wss-summer-school/#/2022/README).

<iframe src="https://docs.google.com/presentation/d/e/2PACX-1vQiyXe-6yn1hfaoibwI8DBYHubQxEd_WEvXte6JqyyjnuY8DXfP64360tRvhmRxOPAoM4Gt6hqKG5ts/embed?start=false&loop=true&delayms=10000" frameborder="0" width="100%" height="400px" allowfullscreen="true" mozallowfullscreen="true" webkitallowfullscreen="true"></iframe>

PDF version of these slides can be downloaded from [here](https://github.com/SPAAM-community/wss-summer-school/raw/main/docs/assets/slides/2022/4c-intro-to-denovoassembly/SPAAM%20Summer%20School%202022%20-%204C%20-%20Genome%20Assembly.pdf).

<iframe width="100%" height="400px" src="https://www.youtube.com/embed/1-o4449Hu4o" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

## Introduction

First of all, what is a **metagenomic** sample? Metagenomic sample is a sample that consists of DNA from more than one source. The number and the type of sources might vary widely between different samples. Typical sources for ancient remains are e.g. the host organism and the microbial species. The important part is that we generally do not know the origin of a DNA molecule prior to analysing the sequencing data generated from the DNA library of this sample. In the example presented in **Figure 1**, our metagenomic sample has DNA from three different sources, here coloured in blue, red, and yellow.


```{r out.width = "80%", echo = F, fig.align = "center"}
knitr::include_graphics("assets/images/chapters/denovo-assembly/metagenome_assembly_scheme.png")
```

**Figure 1**: **Overview of the ways how to analyse metagenomic sequencing data.**

How can we determine the sources of the DNA that we have in our metagenomic sample? There are three main options whose _pros_ and _cons_ are summarised in **Table 1**.

**Table 1**: **Pros and cons of the major methods for determining the sources of the DNA.**

```{r results = "asis", echo = F}
tibble(method = c("reference-based alignment", "single-genome assembly", "metagenome assembly"),
       pros = c("highly sensitive, applicable to aDNA samples",
                "high-quality genomes from cultivated bacteria",
                "able to recover unknown diversity present in sample"),
       cons = c("requires all sources to be represented in database",
                "not available for ancient DNA samples",
                "highly dependent on preservation of ancient DNA")) %>%
pandoc.table(., split.tables = Inf)
```

Unitl recently, the only option for ancient DNA samples was to take the short-read sequencing data and align them against some known reference genomes. However, this approach is heavily relying on the whether all sources of our samples are represented in the available reference genomes. If a source is missing in the reference database - in our toy example, this is the case for the yellow source (**Figure 1**) -, then we won't be able to detect it using this reference database.

While there is a potential workaround for modern metagenomic samples, _single-genome assembly_ it relies on being able to cultivate a microbial species to obtain an isolate. This is unfeasible for ancient metagenomic samples because there are no more viable microbial cells available that could be cultivated.

Around 2015, a technical revolution started when the first programs, such as MEGAHIT [@LiMegahit2015] and metaSPAdes [@Nurk2017], were published that could successfully perform *de novo* assembly from metagenomic data. Since then, tens of thousands metagenomic samples have been assembled and it was revealed that even well studied environments, such as the human gut microbiome, have a lot of additional microbial diversity that could not be discovered via culturing and isolation [@Almeida2021]. 

The technical advancement of now being able to perform *de novo* assembly on metagenomic samples led to an explosion of studies that analysed samples that were considered almost impossible to study beforehand. For researchers that are exposed to ancient DNA, the imminent question arises: can we apply the same methods to ancient DNA data? In this practical, we will walk through all required steps that are necessary to successfully perform *de novo* assembly from ancient DNA metagenomic sequencing data and show you what you can do once you have obtained the data.

## Practical

### Sample overview

For this practical, I selected a palaeofaeces sample from the study by @Maixner2021, who generated deep metagenomic sequencing data for four palaeofaeces samples that were excavated from an Austrian salt mine in Hallstatt and were associated with the Celtic Iron Age. We will focus on the youngest sample, **2612**, which was dated to be just a few hundred years old (**Figure 2**).

```{r out.width = "60%", echo = F, fig.align = "center"}
knitr::include_graphics("assets/images/chapters/denovo-assembly/Maixner2021_GraphicalAbstract.jpg")
```

**Figure 2**: **The graphical abstract of Maixner et al. (2021).**

However, because the sample was very deeply sequenced for more than 250 million paired-end reads and we do not want to wait for days for the analysis to finish, we will not use all data but just a sub-sample of them.

You can access the sub-sampled data by changing to the folder 

```bash
cd /path/to/de_novo_assembly
```

on the cluster. If you would like to repeat the practical on your own computational infrastructure, you can download the sequencing data from here:

```{bash, eval = F}
wget https://share.eva.mpg.de/index.php/s/CtLq2R9iqEcAFyg/download/2612_R1.fastq.gz
wget https://share.eva.mpg.de/index.php/s/mc5JrpDWdL4rC24/download/2612_R2.fastq.gz
```

::: {.callout-tip title="Question" appearence="simple"}

**How many sequences are in each FastQ file?**

Hint: You can run either `seqtk size <FastQ file>` or `bioawk -c fastx 'END{print NR}' <FastQ file>` to find this out.

:::

::: {.callout-note collapse="true" title="Answer"}
There are about 3.25 million paired-end sequences in these files.
:::

### Preparing the sequencing data for _de novo_ assembly

Before running the actual assembly, we need to pre-process our sequencing data. Typical pre-processing steps include the trimming of adapter sequences and barcodes from the sequencing data and the removal of host or contaminant sequences, such as the bacteriophage PhiX, which is commonly used sequenced as a quality control.

Many assembly pipelines, such as [nf-core/mag](https://nf-co.re/mag), have these steps automatically included, however, these steps can be performed prior to it, too. For this practical, I have performed these steps for us and we could directly continue with the _de novo_ assembly.

::: {.callout-important}
The characteristic of ancient DNA samples that pre-determines the success of the _de novo_ assembly is the **distribution of the DNA molecule length**. Determine this distribution prior to running the _de novo_ assembly to be able to predict the results of the _de novo_ assembly.
:::

However, since the average length of the insert size of sequencing data (**Figure 3**) is highly
correlated with the success of the assembly, we want to first evaluate it. For this we can run make
use of the program `fastp` [@Chen2018].

```{r out.width = "60%", echo = F, fig.align = "center"}
knitr::include_graphics("assets/images/chapters/denovo-assembly/insert_size_scheme.png")
```

**Figure 3**: **Scheme of a read pair of Illumina sequencing data.**

Fastp will try to overlap the two mates of a read pair and, if this is possible, return the length of the merged sequence, which is identical to insert size or the DNA molecule length. If the two mates cannot be overlapped, we will not be able to know the exact DNA molecule length but only know that it is longer than 290 bp (each read has a length of 150 bp and FastP requires a 11 bp overlap between the reads).

The final histogram of the insert sizes that is returned by FastP can tell us how well preserved the DNA of an ancient sample is (**Figure 4**). The more the distribution is skewed to the right, i.e. the longer the DNA molecules are, the more likely we are to obtain long contiguous DNA sequences from the _de novo_ assembly. A distribution that is skewed to the left means that the DNA molecules are more degraded and this lowers our chances for obtaining long continuous sequences.

```{r out.width = "70%", echo = F, fig.align = "center"}
knitr::include_graphics("assets/images/chapters/denovo-assembly/Orlando2013_FigS4.5.png")
```

**Figure 4**: **Example of a DNA molecule length distribution of a well-preserved ancient DNA
sample.** This histogram belongs to a 700,000-year-old horse that was preserved in permafrost
(@Orlando2013 Fig. S4).

To infer the distribution of the DNA molecules, we can run the command

```{bash, eval = F}
fastp --in1 2612_R1.fastq.gz \
      --in2 2612_R2.fastq.gz \
      --stdout --merge -A -G -Q -L --json /dev/null --html overlaps.html \
> /dev/null
```

::: {.callout-tip title="Question" appearence="simple"}

**Infer the distribution of the DNA molecule length of the sequencing data. Is this sample well-preserved?**

Hint: You can easily inspect the distribution by opening the HTML report `overlaps.html`.

:::

::: {.callout-note collapse="true" title="Answer"}

Here is the histogram of the insert sizes determined by _fastp_. By default, _fastp_ will only keep
reads that are longer than 30 bp and requires an overlap between the read mates of 30 bp. The
maximum read length is 150 bp, therefore, the histogram only spreads from 31 to 271 bp in total.

```{r out.width = "70%", echo = F, fig.align = "center"}
knitr::include_graphics("assets/images/chapters/denovo-assembly/2612_insertsize_hist.png")
```

The sequencing data for the sample **2612** were generated across eight different sequencing runs,
which differed in their nominal length. Some sequencing runs were 2x 100 bp, while others were 2x
150 bp. This is the reason why we observe two peaks just short of 100 and 150 bp. The difference to
the nominal length is caused by the quality trimming of the data.

Overall, we have almost no short DNA molecules (< 50 bp) but most DNA molecules are longer than 80
bp. Additionally, there were > 200,000 read pairs that could not be overlapped. Therefore, we can
conclude that the sample **2612** is moderately degraded ancient DNA sample and has many long DNA
molecules.
:::


### _De novo_ assembly

