---
title: Introduction to _de novo_ Genome Assembly
author: Alexander HÃ¼bner
---

```{r echo = F, message = F}
library(tidyverse)
library(pander)
```


::: {.callout-tip}
For this chapter's exercises, if not already performed, you will need to create the [conda environment](before-you-start.qmd#creating-a-conda-environment) from the `yml` file in the following [link](https://github.com/SPAAM-community/intro-to-ancient-metagenomics-book/raw/main/assets/envs/denovo-assembly.yml) (right click and save as to download), and once created, activate the environment with:

```bash
conda activate denovo-assembly
```
:::

### Lecture

Lecture slides and video from the [2022 edition of the summer school](https://www.spaam-community.org/wss-summer-school/#/2022/README).

<iframe src="https://docs.google.com/presentation/d/e/2PACX-1vQiyXe-6yn1hfaoibwI8DBYHubQxEd_WEvXte6JqyyjnuY8DXfP64360tRvhmRxOPAoM4Gt6hqKG5ts/embed?start=false&loop=true&delayms=10000" frameborder="0" width="100%" height="400px" allowfullscreen="true" mozallowfullscreen="true" webkitallowfullscreen="true"></iframe>

PDF version of these slides can be downloaded from [here](https://github.com/SPAAM-community/wss-summer-school/raw/main/docs/assets/slides/2022/4c-intro-to-denovoassembly/SPAAM%20Summer%20School%202022%20-%204C%20-%20Genome%20Assembly.pdf).

<iframe width="100%" height="400px" src="https://www.youtube.com/embed/1-o4449Hu4o" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

## Introduction

First of all, what is a **metagenomic** sample? Metagenomic sample is a sample that consists of DNA from more than one source. The number and the type of sources might vary widely between different samples. Typical sources for ancient remains are e.g. the host organism and the microbial species. The important part is that we generally do not know the origin of a DNA molecule prior to analysing the sequencing data generated from the DNA library of this sample. In the example presented in **Figure 1**, our metagenomic sample has DNA from three different sources, here coloured in blue, red, and yellow.


```{r out.width = "80%", echo = F, fig.align = "center"}
knitr::include_graphics("assets/images/chapters/denovo-assembly/metagenome_assembly_scheme.png")
```

**Figure 1**: **Overview of the ways how to analyse metagenomic sequencing data.**

How can we determine the sources of the DNA that we have in our metagenomic sample? There are three main options whose _pros_ and _cons_ are summarised in **Table 1**.

**Table 1**: **Pros and cons of the major methods for determining the sources of the DNA.**

```{r results = "asis", echo = F}
tibble(method = c("reference-based alignment", "single-genome assembly", "metagenome assembly"),
       pros = c("highly sensitive, applicable to aDNA samples",
                "high-quality genomes from cultivated bacteria",
                "able to recover unknown diversity present in sample"),
       cons = c("requires all sources to be represented in database",
                "not available for ancient DNA samples",
                "highly dependent on preservation of ancient DNA")) %>%
pandoc.table(., split.tables = Inf)
```

Unitl recently, the only option for ancient DNA samples was to take the short-read sequencing data and align them against some known reference genomes. However, this approach is heavily relying on the whether all sources of our samples are represented in the available reference genomes. If a source is missing in the reference database - in our toy example, this is the case for the yellow source (**Figure 1**) -, then we won't be able to detect it using this reference database.

While there is a potential workaround for modern metagenomic samples, _single-genome assembly_ it relies on being able to cultivate a microbial species to obtain an isolate. This is unfeasible for ancient metagenomic samples because there are no more viable microbial cells available that could be cultivated.

Around 2015, a technical revolution started when the first programs, such as MEGAHIT [@LiMegahit2015] and metaSPAdes [@Nurk2017], were published that could successfully perform *de novo* assembly from metagenomic data. Since then, tens of thousands metagenomic samples have been assembled and it was revealed that even well studied environments, such as the human gut microbiome, have a lot of additional microbial diversity that could not be discovered via culturing and isolation [@Almeida2021]. 

The technical advancement of now being able to perform *de novo* assembly on metagenomic samples led to an explosion of studies that analysed samples that were considered almost impossible to study beforehand. For researchers that are exposed to ancient DNA, the imminent question arises: can we apply the same methods to ancient DNA data? In this practical, we will walk through all required steps that are necessary to successfully perform *de novo* assembly from ancient DNA metagenomic sequencing data and show you what you can do once you have obtained the data.

## Practical

### Sample overview

For this practical, I selected a palaeofaeces sample from the study by @Maixner2021, who generated deep metagenomic sequencing data for four palaeofaeces samples that were excavated from an Austrian salt mine in Hallstatt and were associated with the Celtic Iron Age. We will focus on the youngest sample, **2612**, which was dated to be just a few hundred years old (**Figure 2**).

```{r out.width = "60%", echo = F, fig.align = "center"}
knitr::include_graphics("assets/images/chapters/denovo-assembly/Maixner2021_GraphicalAbstract.jpg")
```

**Figure 2**: **The graphical abstract of Maixner et al. (2021).**

However, because the sample was very deeply sequenced for more than 250 million paired-end reads and we do not want to wait for days for the analysis to finish, we will not use all data but just a sub-sample of them.

You can access the sub-sampled data by changing to the folder 

```bash
cd /path/to/de_novo_assembly
```

on the cluster. If you would like to repeat the practical on your own computational infrastructure, you can download the sequencing data from here:

```{bash, eval = F}
wget https://share.eva.mpg.de/index.php/s/CtLq2R9iqEcAFyg/download/2612_R1.fastq.gz
wget https://share.eva.mpg.de/index.php/s/mc5JrpDWdL4rC24/download/2612_R2.fastq.gz
```

::: {.callout-tip title="Question" appearence="simple"}

**How many sequences are in each FastQ file?**

Hint: You can run `seqtk size <FastQ file>` to find this out.

:::

::: {.callout-note collapse="true" title="Answer"}
There are about 32.5 million paired-end sequences in these files.
:::


### _De novo_ assembly

