---
title: Introduction to Ancient DNA
author: Christina Warinner
bibliography: assets/references/introduction-to-ancient-dna.bib
---

::: {.callout-important}
ðŸš§ This page is still under construction ðŸš§
:::

This chapter is still under construction. For the slides and recorded lecture version of this session, please see the Werner Siemens-Stiftung funded SPAAM Summer School: Introduction to Ancient Metagenomics [website](https://www.spaam-community.org/wss-summer-school/#/2023/teaching-material).

--

The field of paleogenomics focuses on the study of ancient DNA, and traces its origins to the 1980s. 
This chapter provides a brief overview of the methodological history of the field and reviews the fundamentals of how ancient DNA is recovered, sequenced, and authenticated today. 
Emphasis is placed on how DNA degrades and how DNA degradation patterns are used to help authenticate ancient DNA at the level of sequences, genomes, and samples, as well as how DNA damage creates both opportunities and downstream challenges when genotyping or generating phylogenies. 

## Ancient DNA beginnings

The origins of the field of paleogenomics can be traced back to 1984 with the publication of the first ancient DNA study by Russell Higuichi and colleagues [@Higuchi1984-zu].
Working in Allan Wilsonâ€™s lab at the University of California at Berkeley, Higuchi investigated DNA from a museum soft tissue specimen of the quagga, a subspecies of zebra that had gone extinct in the late 19th century (@fig-introductiontoancientdna-01quagga). 
Compared to today, 1984 was a very different world in terms of technological capabilities within biology and genetics. To help put this into perspective, @fig-introductiontoancientdna-01quagga shows the DNA sequences that were published in this first paper. 
They consist of partial DNA sequences from two mitochondrial genes â€“ cytochrome oxidase I and a second unidentified gene that was later determined to be NADH dehydrogenase I. 
Sequencing these two short segments of DNA required an enormous amount of time and effort to achieve, and all of the work was analog. 
In fact, most of the digital tools we take for granted today did not yet exist in 1984. 
Journals did not have websites, manuscripts were submitted by post, and genetic data was mostly tabulated and analyzed by hand. Microsoft Excel, which is today a kind of ubiquitous software for basic spreadsheet data manipulation, wasn't even invented and released until 1985 (for Mac) and 1987 (for Windows). 

![The first ancient DNA sequences. In 1984, ancient DNA was successfully sequenced from a museum specimen of muscle tissue from a quagga, an extinct subspecies of zebra. Short DNA sequences from two genes were obtained using Sanger Sequencing, resulting in a total of 226 bp of reconstructed quagga DNA. Quagga image by Frederick York, 1870, Wikipedia, public domain. DNA sequences from [@Higuchi1984-zu].](assets/images/chapters/introduction-to-ancient-dna/fig-introductiontoancientdna-01quagga.png){#fig-introductiontoancientdna-01quagga}

This first ancient DNA study, published at the very beginning of the era of genetic sequencing in the journal _Nature_, illustrates what an important achievement it was in the 1980s to be able to sequence even this small amount of quagga DNA. 
However, the methods they used are very different from the methods we use today. 
Higuchi and colleagues achieved the genetic sequencing reported in their paper using a technique called primed-synthesis dideoxynucleoside chain-termination sequencing, a method that had been developed by Frederick Sanger and colleagues only seven years before in 1977 [@Sanger1977-wk]. 
Sangerâ€™s method, which came to be known as **Sanger Sequencing**, was the first truly successful and useful method for sequencing DNA, and it became the primary method of DNA sequencing for the next 35 years. 
This was a breakthrough moment in the history of genetics, and it is worth exploring how the method works. 

## References



