---
title: Introduction to Ancient DNA
author: Christina Warinner
bibliography: assets/references/introduction-to-ancient-dna.bib
---

::: {.callout-important}
🚧 This page is still under construction 🚧
:::

This chapter is still under construction. For the slides and recorded lecture version of this session, please see the Werner Siemens-Stiftung funded SPAAM Summer School: Introduction to Ancient Metagenomics [website](https://www.spaam-community.org/wss-summer-school/#/2023/teaching-material).

--

The field of paleogenomics focuses on the study of ancient DNA, and traces its origins to the 1980s. 
This chapter provides a brief overview of the methodological history of the field and reviews the fundamentals of how ancient DNA is recovered, sequenced, and authenticated today. 
Emphasis is placed on how DNA degrades and how DNA degradation patterns are used to help authenticate ancient DNA at the level of sequences, genomes, and samples, as well as how DNA damage creates both opportunities and downstream challenges when genotyping or generating phylogenies. 

## Ancient DNA beginnings

The origins of the field of paleogenomics can be traced back to 1984 with the publication of the first ancient DNA study by Russell Higuichi and colleagues [@Higuchi1984-zu].
Working in Allan Wilson’s lab at the University of California at Berkeley, Higuchi investigated DNA from a museum soft tissue specimen of the quagga, a subspecies of zebra that had gone extinct in the late 19th century (@fig-introductiontoancientdna-01quagga). 
Compared to today, 1984 was a very different world in terms of technological capabilities within biology and genetics. To help put this into perspective, @fig-introductiontoancientdna-01quagga shows the DNA sequences that were published in this first paper. 
They consist of partial DNA sequences from two mitochondrial genes – cytochrome oxidase I and a second unidentified gene that was later determined to be NADH dehydrogenase I. 
Sequencing these two short segments of DNA required an enormous amount of time and effort to achieve, and all of the work was analog. 
In fact, most of the digital tools we take for granted today did not yet exist in 1984. 
Journals did not have websites, manuscripts were submitted by post, and genetic data was mostly tabulated and analyzed by hand. Microsoft Excel, which is today a kind of ubiquitous software for basic spreadsheet data manipulation, wasn't even invented and released until 1985 (for Mac) and 1987 (for Windows). 

![The first ancient DNA sequences. In 1984, ancient DNA was successfully sequenced from a museum specimen of muscle tissue from a quagga, an extinct subspecies of zebra. Short DNA sequences from two genes were obtained using Sanger Sequencing, resulting in a total of 226 bp of reconstructed quagga DNA. Quagga image by Frederick York, 1870, Wikipedia, public domain. DNA sequences from [@Higuchi1984-zu].](assets/images/chapters/introduction-to-ancient-dna/fig-introductiontoancientdna-01quagga.png){#fig-introductiontoancientdna-01quagga}

This first ancient DNA study, published at the very beginning of the era of genetic sequencing in the journal _Nature_, illustrates what an important achievement it was in the 1980s to be able to sequence even this small amount of quagga DNA. 
However, the methods they used are very different from the methods we use today. 
Higuchi and colleagues achieved the genetic sequencing reported in their paper using a technique called primed-synthesis dideoxynucleoside chain-termination sequencing, a method that had been developed by Frederick Sanger and colleagues only seven years before in 1977 [@Sanger1977-wk]. 
Sanger’s method, which came to be known as **Sanger Sequencing**, was the first truly successful and useful method for sequencing DNA, and it became the primary method of DNA sequencing for the next 35 years. 
This was a breakthrough moment in the history of genetics, and it is worth exploring how the method works. 

In the previous chapter, you learned some of the basic concepts of DNA sequencing that are used for **next generation sequencing (NGS)**. 
NGS builds upon the basic principles of the original Sanger method, and so it is worth examining the Sanger method in more detail here. 
The Sanger method works by using a DNA polymerase to copy a large pool of identical DNA templates using a mixture of ordinary nucleotides and nucleotides containing a blocking component. 
As the polymerase extends along each DNA template molecule, it incorporates the available nucleotides, and when it incorporates a blocking nucleotide it prematurely stops. 
The random incorporation of blocking nucleotides at different points along the template molecules results in DNA of different extension lengths, which can then be separated and ordered by size using gel electrophoresis. 
In the original Sanger method, each round of sequencing required four separate reactions - each containing blocking nucleotides for a different base. 
Following gel electrophoresis, the different banding patterns of the four reactions could then be used to determine the base present at each particular position in the sequence, and thus determining the overall DNA sequence of the template. 

@fig-introductiontoancientdna-02sangerhorse shows an example of how this works from another early ancient DNA study by Matthias Höss and Svante Pääbo on a different extinct equid [@Hoss1993-sm]. 
The bands are “read” from bottom to top, which represents the 5’ to 3’ orientation of the template molecule. 
The four lanes of the gel represent the four separate polymerase extension reactions, each containing a mixture of ordinary nucleotides and blocking nucleotides corresponding to the base annotated at the top of each lane. 
The shortest DNA sequences travel fastest through the gel and are visualized as bands at the bottom of the gel; these represent the beginning of the DNA sequence. 
The longest DNA sequences travel more slowly through the gel and are visualized as bands at the top; these represent the end of the DNA sequence. 
Determining the DNA sequence was initially an entirely analogue process generally carried out with a ruler and pencil. 
Band by band, the bases would be recorded from bottom to top, resulting in a consensus sequence. 
This process, setting aside the prior laboratory work necessary to produce sufficient template DNA for sequencing, would itself have taken a full workday in order to set up the reactions, perform the extensions, run the gel, image the gel, develop the film, and then manually read out the DNA sequences. 
Although automated sequencing and basecalling first became available for the Sanger method in 1987 with the release of the Applied Biosystems ABI 370 instrument, which utilized fluorophores instead of radioactive molecules to detect the DNA bases (making it safer) and a personal computer for digital basecalling (making it faster), analog methods continued to be widely used well into the 1990s. 

![Example of ancient DNA sequenced by the Sanger method. In 1993, mitochondrial DNA was successfully sequenced from a 25,000-year-old horse bone using Sanger Sequencing. The portion of the DNA sequence alignment visible on the gel is highlighted in gray. Adapted from [@Hoss1993-sm].](assets/images/chapters/introduction-to-ancient-dna/fig-introductiontoancientdna-02sangerhorse.png){#fig-introductiontoancientdna-02sangerhorse}

Remarkably, the original quagga sequences were obtained using Sanger Sequencing without the use of **polymerase chain reaction (PCR)**. 
Although PCR was invented by Kary Mullis in the 1980s ([@Saiki1985-bk], [@Saiki1988-mc]), it did not become widely accessible until the early 1990s ([@Bartlett2003-gb], [@Mullis1992-of]). 
Thus Higuchi and colleagues achieved the first ancient DNA sequences through even more laborious methods based on restriction enzymes and bacterial cloning. 
In total, their study reports two ancient DNA sequences, one 114 bp long and the other 112 bp long, from one ancient sample. 
Now contrast this to today, when it is possible to routinely generate >50 billion high quality ancient DNA sequences every 48 hours on an Illumina NovaSeq X instrument (https://emea.illumina.com/systems/sequencing-platforms/novaseq-x-plus.html). 
 
Ancient DNA sequencing has come a long way since the 1980s, and grappling with these changes is the rationale for developing this textbook and companion course on Ancient Metagenomics, which focuses on bioinformatic coding and scripting. 
During the Sanger sequencing era, from the 1980s to the 2000s, the vast majority of the hours of work that went into ancient DNA analysis was in the laboratory and consisted of extensive wet lab preparation of samples to reach the point where a relatively short DNA sequence could be generated. 
It is no wonder then that this era demanded intensive training in biochemistry and molecular biology. 
The situation is very different today, in which the laboratory methods have become highly standardized but the data output is enormous. 
Since the rise of high-throughput sequencing in 2010s, the main challenge for ancient DNA researchers has become how to manage this deluge of genetic data, and this requires specialized computational and bioinformatics skills to be able to handle, process, and interpret the vast amounts of data that make up today’s ancient DNA datasets. 


## References



