---
title: Introduction to Ancient DNA
author: Christina Warinner
bibliography: assets/references/introduction-to-ancient-dna.bib
---

::: {.callout-important}
🚧 This page is still under construction 🚧
:::

This chapter is still under construction. For the slides and recorded lecture version of this session, please see the Werner Siemens-Stiftung funded SPAAM Summer School: Introduction to Ancient Metagenomics [website](https://www.spaam-community.org/wss-summer-school/#/2023/teaching-material).

--

The field of paleogenomics focuses on the study of ancient DNA, and traces its origins to the 1980s. 
This chapter provides a brief overview of the methodological history of the field and reviews the fundamentals of how ancient DNA is recovered, sequenced, and authenticated today. 
Emphasis is placed on how DNA degrades and how DNA degradation patterns are used to help authenticate ancient DNA at the level of sequences, genomes, and samples, as well as how DNA damage creates both opportunities and downstream challenges when genotyping or generating phylogenies. 

## Ancient DNA beginnings

The origins of the field of paleogenomics can be traced back to 1984 with the publication of the first ancient DNA study by Russell Higuichi and colleagues [@Higuchi1984-zu].
Working in Allan Wilson’s lab at the University of California at Berkeley, Higuchi investigated DNA from a museum soft tissue specimen of the quagga, a subspecies of zebra that had gone extinct in the late 19th century (@fig-introductiontoancientdna-01quagga). 
Compared to today, 1984 was a very different world in terms of technological capabilities within biology and genetics. To help put this into perspective, @fig-introductiontoancientdna-01quagga shows the DNA sequences that were published in this first paper.
They consist of partial DNA sequences from two mitochondrial genes – cytochrome oxidase I and a second unidentified gene that was later determined to be NADH dehydrogenase I. 
Sequencing these two short segments of DNA required an enormous amount of time and effort to achieve, and all of the work was analog. 
In fact, most of the digital tools we take for granted today did not yet exist in 1984. 
Journals did not have websites, manuscripts were submitted by post, and genetic data was mostly tabulated and analyzed by hand. Microsoft Excel, which is today a kind of ubiquitous software for basic spreadsheet data manipulation, wasn't even invented and released until 1985 (for Mac) and 1987 (for Windows). 

![The first ancient DNA sequences. In 1984, ancient DNA was successfully sequenced from a museum specimen of muscle tissue from a quagga, an extinct subspecies of zebra. Short DNA sequences from two genes were obtained using Sanger Sequencing, resulting in a total of 226 bp of reconstructed quagga DNA. Quagga image by Frederick York, 1870, Wikipedia, public domain. DNA sequences from [@Higuchi1984-zu].](assets/images/chapters/introduction-to-ancient-dna/fig-introductiontoancientdna-01quagga.png){#fig-introductiontoancientdna-01quagga}

This first ancient DNA study, published at the very beginning of the era of genetic sequencing in the journal _Nature_, illustrates what an important achievement it was in the 1980s to be able to sequence even this small amount of quagga DNA. 
However, the methods they used are very different from the methods we use today. 
Higuchi and colleagues achieved the genetic sequencing reported in their paper using a technique called primed-synthesis dideoxynucleoside chain-termination sequencing, a method that had been developed by Frederick Sanger and colleagues only seven years before in 1977 [@Sanger1977-wk]. 
Sanger’s method, which came to be known as **Sanger Sequencing**, was the first truly successful and useful method for sequencing DNA, and it became the primary method of DNA sequencing for the next 35 years. 
This was a breakthrough moment in the history of genetics, and it is worth exploring how the method works. 

In the previous chapter, you learned some of the basic concepts of DNA sequencing that are used for **next generation sequencing (NGS)**. 
NGS builds upon the basic principles of the original Sanger method, and so it is worth examining the Sanger method in more detail here. 
The Sanger method works by using a DNA polymerase to copy a large pool of identical DNA templates using a mixture of ordinary nucleotides and nucleotides containing a blocking component. 
As the polymerase extends along each DNA template molecule, it incorporates the available nucleotides, and when it incorporates a blocking nucleotide it prematurely stops. 
The random incorporation of blocking nucleotides at different points along the template molecules results in DNA of different extension lengths, which can then be separated and ordered by size using gel electrophoresis. 
In the original Sanger method, each round of sequencing required four separate reactions - each containing blocking nucleotides for a different base. 
Following gel electrophoresis, the different banding patterns of the four reactions could then be used to determine the base present at each particular position in the sequence, and thus determining the overall DNA sequence of the template. 

@fig-introductiontoancientdna-02sangerhorse shows an example of how this works from another early ancient DNA study by Matthias Höss and Svante Pääbo on a different extinct equid [@Hoss1993-sm]. 
The bands are “read” from bottom to top, which represents the 5’ to 3’ orientation of the template molecule. 
The four lanes of the gel represent the four separate polymerase extension reactions, each containing a mixture of ordinary nucleotides and blocking nucleotides corresponding to the base annotated at the top of each lane. 
The shortest DNA sequences travel fastest through the gel and are visualized as bands at the bottom of the gel; these represent the beginning of the DNA sequence. 
The longest DNA sequences travel more slowly through the gel and are visualized as bands at the top; these represent the end of the DNA sequence. 
Determining the DNA sequence was initially an entirely analogue process generally carried out with a ruler and pencil. 
Band by band, the bases would be recorded from bottom to top, resulting in a consensus sequence. 
This process, setting aside the prior laboratory work necessary to produce sufficient template DNA for sequencing, would itself have taken a full workday in order to set up the reactions, perform the extensions, run the gel, image the gel, develop the film, and then manually read out the DNA sequences. 
Although automated sequencing and basecalling first became available for the Sanger method in 1987 with the release of the Applied Biosystems ABI 370 instrument, which utilized fluorophores instead of radioactive molecules to detect the DNA bases (making it safer) and a personal computer for digital basecalling (making it faster), analog methods continued to be widely used well into the 1990s. 

![Example of ancient DNA sequenced by the Sanger method. In 1993, mitochondrial DNA was successfully sequenced from a 25,000-year-old horse bone using Sanger Sequencing. The portion of the DNA sequence alignment visible on the gel is highlighted in gray. Adapted from [@Hoss1993-sm].](assets/images/chapters/introduction-to-ancient-dna/fig-introductiontoancientdna-02sangerhorse.png){#fig-introductiontoancientdna-02sangerhorse}

Remarkably, the original quagga sequences were obtained using Sanger Sequencing without the use of **polymerase chain reaction (PCR)**. 
Although PCR was invented by Kary Mullis in the 1980s [@Saiki1985-bk;@Saiki1988-mc], it did not become widely accessible until the early 1990s [@Bartlett2003-gb;@Mullis1992-of]. 
Thus Higuchi and colleagues achieved the first ancient DNA sequences through even more laborious methods based on restriction enzymes and bacterial cloning. 
In total, their study reports two ancient DNA sequences, one 114 bp long and the other 112 bp long, from one ancient sample. 
Now contrast this to today, when it is possible to routinely generate >50 billion high quality ancient DNA sequences every 48 hours on an Illumina NovaSeq X instrument (https://emea.illumina.com/systems/sequencing-platforms/novaseq-x-plus.html). 
 
Ancient DNA sequencing has come a long way since the 1980s, and grappling with these changes is the rationale for developing this textbook and companion course on Ancient Metagenomics, which focuses on bioinformatic coding and scripting. 
During the Sanger sequencing era, from the 1980s to the 2000s, the vast majority of the hours of work that went into ancient DNA analysis was in the laboratory and consisted of extensive wet lab preparation of samples to reach the point where a relatively short DNA sequence could be generated. 
It is no wonder then that this era demanded intensive training in biochemistry and molecular biology. 
The situation is very different today, in which the laboratory methods have become highly standardized but the data output is enormous. 
Since the rise of high-throughput sequencing in 2010s, the main challenge for ancient DNA researchers has become how to manage this deluge of genetic data, and this requires specialized computational and bioinformatics skills to be able to handle, process, and interpret the vast amounts of data that make up today’s ancient DNA datasets. 

## From quagga to ancient microbes

How did we get from quagga to ancient microbes? 
It has been an incredible journey, made all the more remarkable by the fact that nearly everything we now know about ancient microbes has been achieved through the study of ancient DNA. 
The rest of this chapter will focus on how genetic technologies are used to study ancient microbes. 
But first, let’s start with some important questions: Where exactly does one find the DNA of ancient microbes? 
How does one archaeologically recover organisms that are too small to see? 
It turns out there are several reliable sources of ancient microbes that have proven to be productive and informative about the past. 
Here, we will review six sources that are particularly important in archaeology: teeth, bones, historic medical specimens, paleofeces, cultural objects, and sediments (@fig-introductiontoancientdna-03sources). 

![Sources of ancient microbial DNA. (a) teeth; (b) pathological bone; (c) historic soft tissue medical specimens; (d) paleofeces; (e) cultural artifact residues; (f) sediments. Images a-b and d-f are courtesy of the author. Images a-b, d-f are courtesy of the author. Image c depicts a detail of an 1897 medical illustration of liver hepatitis; the image is licensed under a Creative Commons Attribution 4.0 International license and comes from Wellcome Images (image 577059i), a website operated by Wellcome Trust, a global charitable foundation based in the United Kingdom.](assets/images/chapters/introduction-to-ancient-dna/fig-introductiontoancientdna-03sources.jpg){#fig-introductiontoancientdna-03sources}

### Teeth

Perhaps the single greatest source of ancient microbial DNA is teeth. 
As the only part of the skeleton normally visible during life, teeth are also the only part of the skeleton typically colonized by microbes. 
@fig-introductiontoancientdna-03sources shows the mandible (lower jaw) of a woman who died around a thousand years ago at the medieval monastic site of Dalheim in Germany [@Warinner2014-hn;@Radini2019-ls]. 
Several important microbially-associated features are visible. First, there are mineral accretions on the surface of her teeth. 
These accretions are called dental calculus, and they form through the periodic calcification of dental plaque [@Fagernas2023-wx;@Warinner2015-qu]. 
Below the calculus, the alveolar bone (which holds the teeth) is thickened, recessed, and porous - indications that the periodontium (the soft and hard tissues supporting the teeth) was chronically inflamed by dental plaque bacteria during life. 
Left untreated, this inflammation has caused partial destruction of the alveolar bone, a medical condition known as periodontitis. 
Thus, even from a simple visual inspection we can already see signs of ancient microbial activity that occurred during this woman’s life. 

@fig-introductiontoancientdna-04teeth shows a closer view of one of the teeth in cross-section using scanning electron microscopy (SEM). 
There are three main sources of microbial DNA that can be obtained from this tooth. 
On the surface of the enamel, which appears as white, the mineral accretions of dental calculus are visible. 
Within the dental calculus are thousands, if not millions, of individual bacterial cells that were calcified in situ (@fig-introductiontoancientdna-04teeth, panel A). 
They are, in a sense, frozen in time, along with their DNA [@Fagernas2023-wx]. 
The process by which dental plaque calcifies into dental calculus happens at regular intervals over an individual’s lifetime, building up a calcified biofilm that can become several millimeters thick. 
This process occurs during life, and in fact dental calculus is the only part of your body that routinely fossilizes while you are still alive. 
What you see in @fig-introductiontoancientdna-04teeth are the calcified remains of the oral microbiome, and specifically the dental plaque bacteria that grow on the surfaces of teeth. 
Although most bacteria within dental calculus are commensals, respiratory and other pathogens affecting the oral cavity have also been identified within calculus, including Klebsiella pneumoniae [@Austin2022-tx] and Mycobacterium leprae [@Fotakis2020-uc]. 

![Archaeological tooth in cross-section and magnified views of different sources of microbial DNA. (a) Semi-fossilized oral microbiome bacteria in dental calculus; (b) pathogen DNA preserved on the walls of the dental pulp chamber; (c) necrobiome bacteria contributing to the decomposition of tooth dentine. Images produced using scanning electron microscopy (SEM) [@Warinner2022-mm].](assets/images/chapters/introduction-to-ancient-dna/fig-introductiontoancientdna-04teeth.jpg){#fig-introductiontoancientdna-04teeth}

@fig-introductiontoancientdna-04teeth (panel B) shows a close-up view of the pulp chamber of the tooth. 
During life, the pulp chamber of the tooth is vascularized. 
There are blood vessels that run through the root canal and feed the dental pulp, which is at the center of each tooth. 
This connects the teeth to the broader circulatory system. 
During life, if a person has a blood-borne infection, any pathogens that are circulating in the blood will also circulate through the teeth in the dental pulp chamber. 
If this person dies while the infection is active, the pathogens will remain in the tooth and decay in place, and their DNA will end up becoming smeared along the walls of the dental pulp chamber. 
The dental pulp chamber is one of the best known sources of DNA from pathogens involved in infectious blood-borne diseases around the time of death [@Spyrou2019-vg]. 
Several pathogens have been identified archaeologically by analyzing ancient DNA in the dental pulp chamber, including Yersinia pestis (Bos et al., 2011), Salmonella enterica Paratyphi C [@Vagene2018-ti], smallpox [@Muhlemann2020-gb], and hepatitis B virus [@Kocher2021-ab].
 
@fig-introductiontoancientdna-04teeth (panel C) shows a part of the tooth dentine that is actively undergoing decomposition. 
This decomposition appears as discoloration in the SEM, and tissue degradation gives the dentine a ragged appearance. 
Such changes are caused by the activity of the necrobiome, the bacteria that contribute to the decay and decomposition of the teeth. 
Within the figure, many small dark bacterial cells are visible. 
They are dark in color because they are not mineralized, and they are not mineralized because they are alive. 
These are the many living bacteria that are slowly breaking down and decomposing the tooth. 
Most necrobiome bacteria are environmental bacteria that originate from the burial sediments [@Warinner2014-hn], but in some cases oral bacteria can also contribute to decomposition [@Mann2018-bs]. 
One thing that is important to keep in mind about the necrobiome is that these bacteria can also be quite old. 
Bacteria invade the body and start to break it down within hours or days of death. When these bacteria die, their DNA will also accumulate damage and appear ancient. 
When analyzing the necrobiome, it is typical to observe a mixture of DNA from bacteria of different ages. Some of the necrobiome DNA will originate from dead bacteria that contributed to the early stages of decomposition, and these dead bacteria may be nearly as old as the tooth itself and will show signs of DNA damage. 
Other necrobiome DNA will originate from living bacteria, including those that have more recently colonized the tooth. 

### Bone 
Bone, and especially pathological bone, is another important source of ancient microbial DNA (@fig-introductiontoancientdna-03sources, panel B). 
Some diseases that infect the skeleton, especially those that are chronic and progress over many months or years, may cause lesions or other characteristic alterations in bone. 
Chronic pulmonary infections, such as tuberculosis, can escape the lungs and infect neighboring tissues, forming lesions on the interior of the ribs or infecting the thoracic vertebrae. 
Invasion of the vertebrae by the bacterial pathogen Mycobacterium tuberculosis can cause large lesions and cavities to form within the vertebral bodies (@fig-introductiontoancientdna-03sources, panel B), which weakens and destabilizes the spine. 
This can then result in the vertebrae collapsing forward, a condition called kyphosis. 
Kyphosis is responsible for the characteristic hunching of the back that accompanies advanced spinal tuberculosis, and it is depicted in a wide range of ancient art. 
Examples of spinal tuberculosis causing kyphosis have been described at the Chiribaya sites along the southern coast of Peru dating to ca. 1000 ya [@Bos2014-mr]. 
Leprosy is another disease that produces characteristic alterations to the skeleton. 
Leprosy is caused by Mycobacterium leprae, and as the disease progresses it results in a widening of the nasal aperture, destruction of the bone around the maxillary dentition, and the destruction and loss of bones in the extremities, especially fingers and toes. 
These features can be recognized osteologically in the archaeological record and have led to the recovery of M. leprae DNA from skeletons across Eurasia [@Schuenemann2018-zd].

### Historic medical specimens 
Historic medical specimens represent a large and varied source of ancient microbial DNA (@fig-introductiontoancientdna-03sources, panel C). 
During the 18th, 19th, and early 20th centuries, many hospitals and medical schools amassed large collections of pathological specimens, which range from biopsies to complete organs. 
Such specimens were typically preserved and stored in alcohol or formalin, although precise documentation of their treatment is often lacking. 
Specimens from the early and mid-20th century also include histology slides and tissue blocks, often formalin-fixed and paraffin embedded (FFPE). 
RNA and DNA can be recovered from such specimens, although success rates are often low due to DNA cross-linking caused by the formalin treatment [@Gryseels2020-mq;@Stiller2016-al]. 
Investigation of historic medical specimens, such as vaccination kits, have led to important discoveries regarding the diversity of vaccinia virus and orthopoxvirus strains used in 19th century smallpox vaccination efforts [@Duggan2020-um], and the study of ancient RNA in FFPE tissue blocks has shown that HIV was locally circulating in Central Africa prior to 1960, more than two decades before its global outbreak [@Gryseels2020-mq;@Worobey2008-cx;@Zhu1998-so]. 

### Paleofeces 
Paleofeces are an important source of ancient microbial DNA relating to the gut microbiome and gastrointestinal pathogens (@fig-introductiontoancientdna-03sources, panel D). 
Feces do not preserve in most environments, but specific conditions that immobilize or eliminate water, such as freezing temperatures, extreme dryness, high salinity, or mineralization, can lead to the long-term preservation of archaeological paleofeces and coprolites [@Shillito2020-ci]. 
Permafrost, dry caves [@Wibowo2021-sl], and salt mines [@Maixner2021-nt] are among the best environments for the long-term preservation of paleofeces. 
In addition to microbial DNA, paleofeces are also a good source of dietary and parasite DNA. 

### Cultural artifact residues 
Residues within cultural artifacts such as serving dishes and food and beverage containers are potential sources of ancient culinary bacteria (@fig-introductiontoancientdna-03sources, panel E). 
For example, food residues from baskets and ceramic bowls in western China have yielded molecular evidence for a variety of yeasts and lactic acid bacteria involved in the making of dairy products [@Xie2016-jb] and sourdough bread [@Shevchenko2014-dv]. 

### Sediments 
Finally, archaeological sediments are a rich source of environmental DNA (eDNA) containing the genetic remains of ancient microbes (@fig-introductiontoancientdna-03sources, panel F). 
Sediment samples can be collected through coring, and resin impregnated sediment blocks previously prepared for micromorphology analysis have also been shown to successfully yield ancient DNA [@Massilani2022-dg]. 
The vast majority of research conducted on ancient sediments to date has focused on plant and mammalian DNA in order to reconstruct ancient ecosystems [@Linderholm2021-qh;@Capo2022-vv;@Kjaer2022-ij;@Zavala2021-zp]. 
However, interest in characterizing ancient microorganisms, such as plankton, is increasing [@Armbrecht2020-ga], and the analysis of ancient bacteria is within reach [@Fernandez-Guerra2023-nr], although distinguishing between live and dead pools can be challenging [@Ellegaard2020-ai;@Wegner2023-ad].

## Defining ancient DNA 
Having defined the main sources of ancient microbial DNA, we might next ask: What makes something _ancient_? 
What is “ancient DNA” as opposed to just DNA? 
What makes it specifically ancient? 
Ancient DNA can be defined as any DNA from a non-living source that shows evidence of molecular degradation. 
You will notice this definition does not specify a particular range of time. 
That is because ancient DNA is not defined by a fixed age, but rather by its condition. 
For example, 100,000-year-old Neanderthal oral microbiome DNA from dental calculus [@Klapper2023-rr] would obviously qualify as ancient DNA. 
And so would 5,000-year-old hepatitis B virus DNA from teeth [@Kocher2021-ab], 3,000-year-old gut microbiome DNA from paleofeces [@Maixner2021-nt], 600-year-old plague DNA from skeletons [@Bos2014-mr], oral bacteria from 19th century great apes in a museum [@Fellows_Yates2021-yd], vaccinia virus DNA from 19th century medical specimens [@Duggan2020-um], and even leprosy DNA from mid-20th century FFPE tissue blocks [@Blevins2020-yf]. 
These are all examples of ancient DNA because they all show similar types of DNA damage that require special handling to be able to analyze. 
Essentially, ancient DNA is DNA that has undergone specific forms of degradation and damage. 

## Genome basics 
Before going further, let's first take a step back and ask: Why does this matter? 
Why is ancient DNA defined by its state of preservation rather than its chronological age? DNA degradation matters when considering the diverse composition, organization, size, and copy number of the genomes we are trying to reconstruct and understand. 
So before moving on, we will now review some genome basics. 

First, let’s consider viruses. Viral genomes can be made up of DNA or RNA, and they can be single-stranded, double-stranded, or partially double-stranded. 
Their genome can consist of a single molecule of nucleic acid or many molecules, and their genomes can be linear or circular. 
@fig-introductiontoancientdna-05genomes shows the seven different types of viral genomes classified within the Baltimore system of viral classification [@Koonin2021-xi]. 
The genomes of viruses are very diverse in structure, organization, and composition. 

Bacterial genomes are more consistent in their organization [@Bobay2017-hh]. 
A bacterial genome typically consists of one long, continuous circle of double-stranded DNA (@fig-introductiontoancientdna-05genomes), although some bacteria, such as _Streptomyces_, have linear genomes [@Galperin2007-wo;@Hinnebusch1993-qr]. 
A bacterial genome is often referred to as a chromosome, and while this terminology is not universally accepted due to differences in the compaction and staining properties of prokaryotic and eukaryotic genomes, the term **chromosome** is widely used by large data archives and platforms, including members of the International Nucleotide Database Collaboration (INSDC). 
In addition to the main genome, bacteria can also have accessory DNA called **plasmids**, which are typically smaller circles of DNA that can be exchanged with other bacteria. 

The genomes of animals, which are eukaryotes, are more complex. 
Animals have two genomes: a **nuclear genome** and a **mitochondrial genome** (@fig-introductiontoancientdna-05genomes). 
The nuclear genome is composed of multiple linear strings of double-stranded DNA, which are called chromosomes (after the Greek chroma, meaning color, which refers to the staining properties of DNA-protein complexes in eukaryotic chromatin). 
Each chromosome is one very long linear piece of DNA that is tightly coiled around histones and packaged within the cell nucleus. 
Animal cells are usually **diploid**, meaning they have two of each chromosome in their nuclear genome. 
The mitochondrial genome (mitogenome) is made up of circular double-stranded DNA, and multiple mitogenome copies are present in each of the mitochondria, the energy-producing organelles in the cell cytoplasm. 
Each cell moreover contains many mitochondria, resulting in an average of 1,000-5,000 mitogenomes per animal cell, depending on cell type [@Moraes2001-la]. 
The reason mitochondria have circular DNA is because they originate from bacteria that entered into other cells at the beginning of the formation of eukaryotes [@Dyall2004-zi;@Martin2015-cy]. 
While they have undergone substantial genome reduction and transfer of genes to the nuclear genome over time, they themselves have retained the circular organization of their DNA within eukaryotic cells for more than a billion years. 

Plant cells are similar in many ways to animal cells, but they also contain additional genomes (@fig-introductiontoancientdna-05genomes). 
Like animals, plants have a nuclear genome made up of multiple linear strings of double-stranded DNA folded up within their cell nucleus, but unlike animal cells they are often **polyploid**, meaning that they can have many copies of each chromosome in their nuclear genome - typically two (diploid), four (tetraploid), six (hexaploid), or eight (octoploid). 
Like animals, plant cells also have a mitochondrial genome that is circular and double-stranded and present in many copies per cell, with the largest number of copies in root cells [@Preuten2010-yx]. 
However, plants also have additional genomes within other organelles, such as chloroplasts. The plant **chloroplast genome** is double-stranded and partially circular [@Bendich2004-xr]. 
It originates from a photosynthetic cyanobacteria that entered into cells of plant ancestors early in the development of eukaryotes [@Dyall2004-zi;@Martin2015-cy]. 

![Genome basics. There are major differences in the composition, organization, size, and copy number of genomes in viruses, bateria, animal cells, and plant cells.](assets/images/chapters/introduction-to-ancient-dna/fig-introductiontoancientdna-05genomes.png){#fig-introductiontoancientdna-05genomes}

In sum, the term genome is used to refer to a variety of different configurations of nucleic acids within viruses, prokaryotes, and eukaryotes. 
What are the relative genome sizes of these different types of organisms? 
Viruses have the smallest genomes, with most viral genomes falling between about 5 and 100 thousand bp long [@Campillo-Balderas2015-ob]. 
Bacteria are bigger. 
Bacterial genomes are, on average, about 1 to 10 million bp long [@Westoby2021-oi], and their plasmids can vary in size from 1 to 400 thousand bp long [@Thomas2020-tr]. 
Large plasmids tend to be present in fewer copies than small plasmids, and plasmid copy number is usually 1-200 per bacterial cell [@Ilhan2019-pk;@Thomas2020-tr]. 
Overall, eukaryotes have the largest genomes, and whereas viral and bacterial genome sizes tend to scale linearly with their number of genes, no such correlation exists in eukaryotes (Bobay and Ochman 2017), a phenomenon that has been described as the C-value paradox or C-value enigma [@Gregory2005-pz]. 
Among animals, birds and mammals generally have nuclear genomes of about 1 to 6 billion bp long [@Kapusta2017-rd] and smaller mitogenomes that are 15-20 thousand bp long but present in 1,000-5,000 copies per cell [@Boore1999-hw;@Moraes2001-la). 
Plants clock in with an average nuclear genome size of 0.5 to 14 billion base pairs [@Michael2014-tu] and rather large mitogenomes (200 thousand to 2 million bp) and chloroplast genomes (100-200 thousand bp) that are each present in multiple copies per cell [@Bendich1987-na;@Green2011-un;@Morley2017-me;@Preuten2010-yx]. 
It is thus clear that genome sizes vary enormously among these different types of organisms, with viruses and microbes typically having much smaller genomes than animals or plants. 

Despite these trends, there are nevertheless many genomic outliers that have much larger or smaller genomes than expected. 
For example, the world's largest known virus, _Pandoravirus salinus_, has a genome of 2.5 million bases, which puts it firmly within the typical genome size range of bacteria [@Campillo-Balderas2015-ob]. 
The world's smallest known bacterial genome, the endosymbiont _Carsonella ruddi_, is barely larger than a virus at 160 thousand bases long [@Nakabachi2006-ys], while the largest known bacterial genome, _Sorangium cellulosum_, is 13 million bases long [@Schneiker2007-hi]. 
The largest known animal genome is that of the lungfish _Protopterus aethiopicus_, at 130 billion bases [@Hidalgo2017-hh]. 
And among plants, the largest known genome is that of a small flowering plant, _Paris japonica_, at 149 billion bases [@Hidalgo2017-hh]. 
Among eukaryotes, the size of the genome is not related to the complexity of the organism, and currently the organism with the largest known genome is a microscopic single-celled amoeba called _Polychaos dubium_, with an estimated genome size of 670 billion bases [@Parfrey2008-js], although its precise genome size is disputed and difficult to measure [@Hidalgo2017-hh]. 
When trying to reconstruct ancient genomes from short, degraded ancient DNA, the genome size of the target organism matters a lot. 

Next, let’s put this information about genome sizes in context. 
Let's take the human genome as an example because it is the genome with which we are most familiar. 
The haploid size, or C-value, of the human genome is approximately 3 gigabase pairs, so 3 billion bp. 
Our body cells are diploid, so each of our cells contains two copies of the human genome, for a total of 6 billion bp. Humans have 23 chromosome pairs, for a total of 46 chromosomes, and each chromosome ranges in size between 50 and 250 million bp. 
And recall that chromosomes are really just long linear strings of DNA - very, very, very, very long DNA strings. Humans also have a mitogenome, which by comparison is very tiny. 
It is only about 16,500 bases long, but it is present in thousands of copies per cell. 
On average, each cell contains 1,000-5,000 mitogenomes, but a mature human egg cell may contain as many as 1.5 million copies of the mitogenome [@Cecchino2019-ud]. 

Have you ever wondered how chromosome 1 came to be named chromosome 1, and why chromosome 12 is called chromosome 12? 
The chromosomes are numbered in order of their length, or at least the length as they were originally calculated using cytogenetics (@fig-introductiontoancientdna-06karyogram). 
Thus, chromosome 1 is the largest chromosome in the human genome, and it measures nearly 250 million bp long. 
In contrast, the shortest chromosomes in the human genome, chromosomes 21 and 22, are each around 50 million bp long. 

![Karyogram micrograph showing the autosomal and sex chromosomes of a human male. Inset shows micrograph of human mitochondrion at a different scale. Chromosome lengths are from [@Piovesan2019-nv]. Karyogram photo credit: National Human Genome Research Institute; image is in public domain. Mitochondrial TEM photo credit: Louisa Howard; image is in public domain.](assets/images/chapters/introduction-to-ancient-dna/fig-introductiontoancientdna-06karyogram.png){#fig-introductiontoancientdna-06karyogram}

Today, sequencing and analyzing high molecular weight genomic DNA from living cells is relatively straightforward, especially with the recent advent of long-read sequences like PacBio and Oxford Nanopore [@Koren2015-ak;@Kovaka2023-rp]. 
However, ancient DNA is very different from the DNA shown in the karyogram in @fig-introductiontoancientdna-06karyogram. 
Ancient DNA is broken and degraded into thousands, if not millions, of pieces. Let’s take, for example, chromosome 1. 
In life, it is one continuous string of DNA that is 248,956,422 bp long. 
After death, it fragments into pieces that are each approximately 50 bp long. So to use a metaphor, I like to refer to ancient DNA as the world's worst jigsaw puzzle. 
Because if you had to put together a jigsaw puzzle of chromosome 1 from ancient DNA fragments, it would be a puzzle with over 5 million pieces. 
Hopefully this puts into perspective the kinds of challenges researchers face when working with ancient DNA. 

## How DNA degrades
How does DNA degrade? 
What kind of damage accumulates in ancient DNA over time? 
Some forms of DNA damage occur during life, for example during disease progression, immune response, and tissue necrosis. 
Such DNA damage is largely driven by the activity of nucleases, which create a variety of double-stranded breaks [@Bokelmann2020-ok;@Didenko2008-gn;@Didenko2003-kf;@Didenko1996-id;@Harkins2020-tp]. 
Other forms of DNA damage occur after death and are largely driven by chemical changes. 
In 1993, Tomas Lindahl was the first to examine the latter forms of postmortem damage in detail [@Lindahl1993-xc]. 
Lindahl presented a theoretical model of DNA degradation based on the chemical properties of DNA and identified **hydrolytic** and **oxidative** decomposition as the most likely types of damage to accumulate in fossil DNA. He also predicted that conditions of high ionic strength, adsorption to hydroxyapatite, and partial dehydration may contribute to long-term DNA preservation. 
Michael Hofreiter and colleagues expanded Lindahl’s model of ancient DNA degradation in an influential review article in 2001, noting that the empirically observed excess of C->T an G->A miscoding lesions in PCR-amplified ancient DNA likely results from deamination of cytosine residues in ancient DNA template molecules [@Hofreiter2001-rw]. 

The chemical bonds within DNA that are most vulnerable to degradation are indicated with arrows in @fig-introductiontoancientdna-07DNAdamage. Although DNA is vulnerable in many places, subsequent research has shown that some of these bonds undergo much faster rates of degradation than others, and that DNA damage generally proceeds in a predictable sequence.  

![DNA degradation. Model of DNA degradation highlighting chemical bonds that are more susceptible to chemical attack and are thus most likely to show evidence of damage in ancient DNA. Four nucleotides are shown with the sequence: 5’-GACT-3’. The phosphate backbone and deoxyribose sugar parts of the DNA molecule are shown in black. The nitrogenous bases are colored: guanine (G), purple; adenine (A), green; cytosine (C), blue; thymine (T), red. G and A are purines; C and T are pyrimidines. Sites of hydrolytic damage are indicated by black arrows; sites of oxidative damage are indicated by gray arrows. The typical order of damage accumulation at major damage sites (large arrows) is indicated by numbers: (1) depurination; (2) nicking; (3) cytosine deamination. Adapted from [@Hofreiter2001-rw;@Lindahl1993-xc]](assets/images/chapters/introduction-to-ancient-dna/fig-introductiontoancientdna-07DNAdamage.png){#fig-introductiontoancientdna-07DNAdamage}

--

[@Hoss1993-sm]
@fig-introductiontoancientdna-02sangerhorse
![caption](assets/images/chapters/introduction-to-ancient-dna/fig-introductiontoancientdna-XXXXX.png){#fig-introductiontoancientdna-XXX}

## References



